pandas/core/index.py

```diff
@@ -382,6 +390,8 @@ def _shallow_copy(self, values=None, infer=False, **kwargs):
         values : the values to create the new Index, optional
         infer : boolean, default False
             if True, infer the new type of the passed values
+        reset_attributes : boolean, default False
+            if True, reset attributes specified in _reset_attributes
         kwargs : updates the default attributes for this Index
```

Comment:
Tried to handle ``freq`` reset for DTI / keep for PeriodIndex operation in the function. Will consider better way and remove from this PR.